src/scope.rs

```diff
@@ -224,6 +230,82 @@ impl<'s> HandleScope<'s, ()> {
   }
 }
 
+#[derive(Clone, Copy, Debug)]
+pub enum GetDataFromSnapshotError {
+  NoData,
+  TryFromTypeError(TryFromTypeError),
```

Comment:
* Could we call the variant maybe just `TypeError` ?
* There's no tests. We should at least have one test for each of the variants.